Source/WebCore/Modules/compression/CompressionStreamEncoder.cpp

```diff
@@ -36,11 +36,10 @@ namespace WebCore {
 
 ExceptionOr<RefPtr<Uint8Array>> CompressionStreamEncoder::encode(const BufferSource&& input)
 {
-    auto* data = input.data();
-    if (!data)
+    if (!input.data())
         return Exception { ExceptionCode::TypeError, "No data provided"_s };
```

Comment:
Is this really a reachable code path? Do we have test coverage? I’m kind of surprised that we have a “no data” case distinct from “data length zero”.

Reply:
I don't think it is a distinct case? We could probably just check if the size is 0 here, I merely kept the existing code to be safe.

Reply:
This won’t raise an exception for size 0 if somehow we make a `std::span` that has a non-null pointer for that case. I’d feel a lot more confident about this if we had test cases, and I didn’t check if we do. Not that refactoring to `std::span` affects this at all.

Reply:
A BufferSource data() can return a nullptr, which is why I had added the check. An empty UInt8Array() being passed in is valid and has a size of 0, which is why we shouldn't add a check for length here.

There are no test cases that execute this code path from the ones I reviewed.

Reply:
> A BufferSource data() can return a nullptr, which is why I had added the check.

The code below will do fine with a nullptr as long as the length is 0. It just won’t throw an exception. But why should it throw an exception? I don’t see how we can have an opinion about that unless we understand how it’s web-exposed.

> There are no test cases that execute this code path from the ones I reviewed.

We should construct the test case that will. Unless it’s impossible and this is really dead code, in which case we should simplify by removing the unneeded check.

Reply:
I don't see a way to construct a test case that would trigger this exception.

I believe we can remove this check.